### PR TITLE
[Review] Add debian packaging

### PR DESCRIPTION
**Pull request by Pro**
This is based on #2302 and adds the necessary files to create a .deb package. See last commit here, the rest will be gone as soon as #2302 is merged

Debian packages can be created with:

```
cd open62541
dpkg_buildpackage -b
```

I already configured launchpad to do automatic nightly builds so that users can add the PPA and just do a apt install to get the latest master.

https://launchpad.net/~open62541-team

### CODE REVIEW

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -65,12 +74,32 @@ endif()
...
 
+if (WIN32)
+    install(PROGRAMS ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/server_ctt.exe
+            DESTINATION bin
+            RENAME ua_server_ctt.exe)
+else()
+    install(PROGRAMS ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/server_ctt
+            DESTINATION bin
+            RENAME ua_server_ctt)
+endif()
```
Comment (traversaro) [inline examples/CMakeLists.txt]: I guess this if can be substituted with: 
~~~cmake
install(PROGRAMS  $<TARGET_FILE:server_ctt>
        DESTINATION bin
        RENAME ua_server_ctt)
~~~
but to be honest I did not try it.

Comment (Pro) [reply examples/CMakeLists.txt]: Ah, cool! Will try that!
Thanks

**Comment by jpfr** [comment]
Looking in my /usr/include folder on Debian, most packages with more than one include file have a dedicated folder. Can we not clutter /include directly?

I propoe that we
- move our "public" header files to /include/open62541 in our source tree
- install them to /usr/include/open62541
- drop the ua_ prefix on public header files

So we can use
```c
#include <open62541/server.h>
```
for both the examples compiled against the source tree and against the installed library.

**Comment by jpfr** [comment]
👍 for a separate #2309.

Imho the package can be "slimmed".
For discussion of removal:
- Internal headers (that are not stable within a release family)
- Precompiled python bytecode .pyc
- Schemas (internal as well as supplied by the OPC Foundation repository)

**Comment by Pro** [comment]
> We could have a separate open62641-dev package.
So that users can only install the library if they have a package that depends on it.
But not the header files..

That's already set up, there will be three packages:

- open62541 (lib only)
- open62541-dev (the whole content of make install, including header and lib)
- open62541-dbg (debugging symbols)

> Imho the package can be "slimmed".

The package contains all the output files of `make install`. I will post your comment to #2309 since that is related to the installed files

**Pull request merged**